v2/sourcedb-to-spanner/terraform/samples/sharded-bulk-migration/README.md

```diff
@@ -0,0 +1,435 @@
+## Sample Scenario: Sharded MySQL to Spanner production bulk migration
+
+> **_SCENARIO:_** This Terraform example illustrates launching bulk (backfill/historical) migration Dataflow jobs for a
+> Sharded MySQL to Spanner, setting up all the required sharding config files appropriately.
+
+## Terraform permissions
+
+In order to create the resources in this sample,
+the `Service account`/`User account` being used to run Terraform
+should have the
+required [permissions](https://cloud.google.com/iam/docs/manage-access-service-accounts#multiple-roles-console).
+There are two ways to add permissions -
+
+1. Adding pre-defined roles to the service account running Terraform.
+2. Creating a custom role with the granular permissions and attaching it to the
+   service account running Terraform.
+
+### Using custom role and granular permissions (recommended)
+
+Following permissions are required -
+
+```shell
+- compute.globalAddresses.create
+- compute.globalAddresses.createInternal
+- compute.globalAddresses.delete
+- compute.globalAddresses.deleteInternal
+- compute.globalAddresses.get
+- compute.globalOperations.get
+- compute.networks.addPeering
+- compute.networks.get
+- compute.networks.listPeeringRoutes
+- compute.networks.removePeering
+- compute.networks.use
+- compute.routes.get
+- compute.routes.list
+- compute.subnetworks.get
+- compute.subnetworks.list
```

Comment:
Why are these needed?

Reply:
May not be, I added them because they were also present on the live template.

Reply:
Okay - let's assess the permissions required for bulk migration specifically and list the minimal required permission set here. This is what we did for live template as well.